### ISSUE TITLE
[QUESTION] Yace return metrics with zero value for stopped and terminated EC2 instances and stopped RDS

### ISSUE THREAD

**Issue by winadm87** [nerdswords/yet-another-cloudwatch-exporter#1372]
YACE version
v0.56.0

Config file
  jobs:
    - type: rds
      regions: [ eu-west-1 ]
      delay: 0
      period: 60
      length: 300
      dimensionNameRequirements:
        - DBInstanceIdentifier
      metrics:
        - name: CPUUtilization
          statistics: [ Average ]
        - name: DatabaseConnections
          statistics: [ Average ]
    - type: ec2
      regions: [ eu-west-1 ]
      delay: 0
      period: 60
      length: 300
      dimensionNameRequirements:
        - InstanceId
      metrics:
        - name: CPUUtilization
          statistics: [ Average ]
        - name: CPUCreditBalance
          statistics: [ Average ]
          length: 600

Current Behavior
Assume we have some EC2 instances and RDS. As we stop EC2 instance or terminate it - Yace continues to return all the collected metrics with 0 value (for about 5 hours for terminated instances).
As for RDS - after we stop the RDS db - Yace also continues to return all the collected metrics for this stopped db with 0 value.

Expected Behavior
After EC2 instance is stopped/terminated, rds stopped - Yace should stop returning metrics at all, may be except for _info metric.

Steps To Reproduce
Create an EC2 instance, start collecting metrics. Stop the instance - observe the metrics with 0 value. Terminate the instance - observe the metrics with 0 value.

Anything else?
We wonder if this is expected behavior? Maybe some flag/parameter should be passed on any level to prevent such behavior?

**Comment by winadm87** [comment]
A little addition to the situation. If we put 
`addCloudwatchTimestamp: true`
to the metric config, e.g. "- name: CPUCreditBalance" - the exporter continues to export metric on the terminated instance with negative timestamp...
`aws_ec2_cpucredit_balance_average{dimension_instance_id="i-0013b...",region="eu-west-1"} 0 -62135596800000`
So the prometheus can't scrap such a metric with "expected timestamp or new record, got "INVALID"" error.

**Comment by kgeckhart** [comment]
Try setting
```
# Passes down the flag `--recently-active PT3H` to the CloudWatch API. This will only return metrics that have been active in the last 3 hours.
# This is useful for reducing the number of metrics returned by CloudWatch, which can be very large for some services. See AWS Cloudwatch API docs for [ListMetrics](https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_ListMetrics.html) for more details.
[ recentlyActiveOnly: <boolean> ]
```
in your config which should limit this to ~3 hours. This is a really unfortunate CloudWatch behavior. 

Regarding the `addCloudwatchTimestamp` behavior, you should upgrade to v0.57 which contains a bugfix for this, https://github.com/nerdswords/yet-another-cloudwatch-exporter/releases/tag/v0.57.0. The bugfix is to drop this data because it is invalid.

**Comment by winadm87** [comment]
Apparently, the problem was solved after the version update to 0.59
